core/dbt/context/base.py

```diff
@@ -659,48 +659,27 @@ def print(msg: str) -> str:
 
     @contextmember
     @staticmethod
-    def diff_of_two_dicts(dict_a, dict_b):
+    def diff_of_two_dicts(
+        dict_a: Dict[str, List[str]], dict_b: Dict[str, List[str]]
+    ) -> Dict[str, List[str]]:
         """
-        Given two dictionaries:
-            dict_a: {'key_x': ['value_1', 'value_2'], 'key_y': ['value_3']}
-            dict_b: {'key_x': ['value_1'], 'key_z': ['value_4']}
-        Return the same dictionary representation of dict_a MINUS dict_b
-
-        All keys returned will be lower case.
-        """
-
-        dict_diff = {}
-        dict_a = {k.lower(): v for k, v in dict_a.items()}
-        dict_b = {k.lower(): v for k, v in dict_b.items()}
-        for k in dict_a:
-            if k in dict_b:
-                a_lowered = map(lambda x: x.lower(), dict_a[k])
-                b_lowered = map(lambda x: x.lower(), dict_b[k])
-                diff = list(set(a_lowered) - set(b_lowered))
-                if diff:
-                    dict_diff.update({k: diff})
-            else:
-                dict_diff.update({k: dict_a[k]})
-        return dict_diff
-
-    @contextmember
-    @staticmethod
-    def diff_of_two_dicts_no_lower(dict_a, dict_b):
-        """
-        Given two dictionaries:
-            dict_a: {'key_x': ['value_1', 'value_2'], 'key_y': ['value_3']}
-            dict_b: {'key_x': ['value_1'], 'key_z': ['value_4']}
-        Return the same dictionary representation of dict_a MINUS dict_b
-
-        No modifications to dict values.
+        Given two dictionaries of type Dict[str, List[str]]:
+            dict_a = {'key_x': ['value_1', 'VALUE_2'], 'KEY_Y': ['value_3']}
+            dict_b = {'key_x': ['value_1'], 'key_z': ['value_4']}
+        Return the same dictionary representation of dict_a MINUS dict_b,
+        performing a case-insensitive comparison between the strings in each.
+        All keys returned will be in the original case of dict_a.
+            returns {'key_x': ['VALUE_2'], 'KEY_Y': ['value_3']}
         """
 
         dict_diff = {}
+        dict_b_lowered = {k.casefold(): [x.casefold() for x in v] for k, v in dict_b.items()}
         for k in dict_a:
-            if k in dict_b:
-                a_lowered = map(lambda x: x.lower(), dict_a[k])
-                b_lowered = map(lambda x: x.lower(), dict_b[k])
-                diff = list(set(a_lowered) - set(b_lowered))
+            if k.casefold() in dict_b_lowered.keys():
+                diff = []
+                for v in dict_a[k]:
+                    if v.casefold() not in dict_b_lowered[k.casefold()]:
+                        diff.append(v)
```

Comment:
[casefold](https://docs.python.org/3/library/stdtypes.html#str.casefold), FTW 💪

Reply:
oh wow thats super cool haven't heard of casefold before.

Reply:
@nathaniel-may is going to take a look at refactoring this for performance / Pythonic-ness. Not going to consider it a blocker for merging this PR in the meantime.